File: src/main/java/net/volcano/jdacommands/client/CommandAttacher.java
package net.volcano.jdacommands.client;

import lombok.extern.slf4j.Slf4j;
import net.volcano.jdacommands.exceptions.command.CommandCompileException;
import net.volcano.jdacommands.interfaces.CommandClient;
import net.volcano.jdacommands.model.command.annotations.CommandController;
import org.springframework.context.ApplicationContext;
import org.springframework.stereotype.Component;

import java.util.Map;

/**
 * Finds and registers all commands in the application context
 */

@Slf4j
@Component
public class CommandAttacher {
	
	public CommandAttacher(ApplicationContext applicationContext,
	                       CommandClient commandClient) throws CommandCompileException {
		
		for (Map.Entry<String, Object> entry : applicationContext.getBeansWithAnnotation(CommandController.class).entrySet()) {
			String name = entry.getKey();
			Object o = entry.getValue();
			commandClient.registerController(o);
		}
		
	}
}

File: src/main/java/net/volcano/jdacommands/client/CommandAttacher.kt
package net.volcano.jdacommands.client

import net.volcano.jdacommands.interfaces.CommandClient
import net.volcano.jdacommands.model.command.annotations.CommandController
import org.springframework.context.ApplicationContext
import org.springframework.stereotype.Component

/**
 * Finds and registers all commands in the application context
 */
@Component
class CommandAttacher(
	applicationContext: ApplicationContext,
	commandClient: CommandClient
) {

	init {

		applicationContext.getBeansWithAnnotation(CommandController::class.java)
			.forEach { (_, v) ->
				commandClient.registerController(v)
			}

	}
}